### DISCUSSION TITLE
Is there a possibility to change the CSS of the tab-title of a tab that is active?

### DISCUSSION THREAD

**Discussion by ralf3u** [geany/geany#3482] [Q&A]
On an eInk-display in Lubuntu 23.04 I can't see at all which tab is in focus because of the Widget Style Qt Style Cleanlooks, that does not work like in Lubuntu 22.10. I know that I can change in geany.css the CSS of a tab-title when a document was changed but not yet saved.

So, is there a possibility to do the same in geany.css but with a tab-title of a tab that is active?

Thank you so much in advance for help.

**Comment by elextr** [comment]
GTK CSS has a selector "pseudoclass" for [state active for tabs](https://docs.gtk.org/gtk3/css-overview.html#theme-labels-in-the-first-notebook-tab), but the `geany-*` settings for text are state independent, they depend on Geany internal state (file changed, readonly etc) which are not available to GTK and Geany ignores state active.  

But you probably can do it for the rest of the tab that Geany does not control using normal GTK, even if you can't control the text colour by state, a flashing purple and orange striped background should make the tab visible ;-)

**Comment by ralf3u** [comment]
>... even if you can't control the text colour by state, ...

I didn't want to change the color. My idea was to make it bold because if I use the eInk-display in text-mode there is only black and white. Is there a possibility to make it bold?
In CSS it is 'font-weight:bold'. But which kind of value in geany.css?

>..., a flashing purple and orange striped background should make the tab visible ;-)

Unfortunately this is not possible, because if I use the eInk-display in text-mode there is only black and white. So if I would use an orange as background, it will be displayed as black, what means that I would not see the name of the tab-title, because the tab-title is in black. A background color would make sense if I could change also the font-color of the tab-title in white.

**Comment by elextr** [comment]
> because if I use the eInk-display in text-mode there is only black and white.

Oh, I was just looking at colour E-ink devices, so I assumed ... 

> can't control the text colour by state ... Is there a possibility to make it bold?

Well, IIUC the `geany-document-status-*` CSSes are applied by Geany depending on its internal state, but they only set `color` (by default) so in theory they should not affect other properties set by other CSSes, so you should be able to set [other properties](https://docs.gtk.org/gtk3/css-properties.html).

Yay, after many attempts the pseudo class is `checked`, adding:

```css
notebook tab:checked label {
	font-weight: bold;
	font-style: oblique;
	font-size: 200%;
}
```
in `geany.css` should make it obvious which is the current tab :-)

You may want to tone it down a bit.

Also works for sidebar and message window.

**Comment by elextr** [comment]
PS don't forget to restart Geany after every change to `geany.css`, reloading configuration is not guaranteed to work.

**Comment by ralf3u** [comment]
I recognized that if I use bold, then the tabs are a little bit moving when I tab-browse in a loop. That is the reason why I decided to use this:
```
notebook tab:checked label {
	color:white;
	background:black;
}
```
Your answer is so helpful for me. I was prepared to give up Lubuntu 23.04 and to go back to Lubuntu 22.10. The effect is even better than I expected.

I'm so thankful. Thank you so much for this precious help.

**Comment by ralf3u** [comment]
For the sake of completeness I add some information to this issue. At the moment these are the best settings for me:
```
/* design of tabs */
tab {
	padding-top:25px;
	padding-bottom:10px;
	padding-left: 35px;
}

/* design of active tabs (e.g. tab of the editor and tab of the sidebar) */
tab:checked {
	font-weight: bold;
	border-bottom: 5px solid black;
}
```
In my last comment, I wrote that "tabs are a little bit moving" while tab-browsing. This problem was solved in #3771.
